pandas/tests/dtypes/test_inference.py

```diff
@@ -131,6 +131,42 @@ def shape(self):
         return self._values.shape
 
 
+class MockDBCursor:
```

Comment:
Do we really need this or can we just set up a test that assigns this to the cursor?

Reply:
Not sure what you mean, but I could drop this specific of a test and add a test with the rest of the database tests. That would be more of a real-world use of the feature.